src/aat/java/uk/gov/hmcts/ccd/datastore/tests/functional/elasticsearch/ElasticsearchCaseSearchTest.java

```diff
@@ -0,0 +1,157 @@
+                searchCaseForExactMatchAndVerifyResponse("DateTimeField", DATE_TIME);
+            }
+        }
+
+        private void searchCaseForExactMatchAndVerifyResponse(String field, String value) {
+            String jsonSearchRequest = createExactMatchSearchRequest(field, value);
+
+            ValidatableResponse response = searchCaseAsPrivateCaseWorker(jsonSearchRequest);
+
+            verifyExactMatchResponse(response, field, value);
+        }
+
+        private String createExactMatchSearchRequest(String field, String value) {
+            return "{"
+                + "  \"query\": {"
+                + "    \"match\": {"
+                + "    \"data." + field + "\" : \"" + value + "\""
+                + "    }"
+                + "  }"
+                + "}";
+        }
+
+        private void verifyExactMatchResponse(ValidatableResponse response, String field, String expectedValue) {
+            response.body("cases.size()", is(1));
+            response.body("cases[0].case_data." + field, is(expectedValue));
+        }
+    }
+
+    @Nested
+    @DisplayName("Wildcard")
+    class Wildcard {
+
+        @Nested
+        @DisplayName("text field")
+        class TextField {
+
+            @Test
+            @DisplayName("should return case matching wildcard expression on a text field")
+            void shouldReturnCaseForWildcardMatchOnTextField() {
+                String wildcardExpr = TEXT.substring(0, 3).toLowerCase() + "*";
+                searchCaseByWildcardAndVerifyResponse("TextField", wildcardExpr, TEXT);
+            }
+        }
+
+        private void searchCaseByWildcardAndVerifyResponse(String field, String wildcardExpr, String expectedValue) {
+            String jsonSearchRequest = createWildcardSearchRequest(field, wildcardExpr);
+
+            ValidatableResponse response = searchCaseAsPrivateCaseWorker(jsonSearchRequest);
+
+            verifyWildcardMatchResponse(response, field, expectedValue);
+        }
+
+        private String createWildcardSearchRequest(String field, String wildcardExpr) {
+            return "{"
+                + "  \"query\": {"
+                + "    \"wildcard\": {"
+                + "    \"data." + field + "\" : \"" + wildcardExpr + "\""
+                + "    }"
+                + "  }"
+                + "}";
+        }
+
+        private void verifyWildcardMatchResponse(ValidatableResponse response, String field, String expectedValue) {
+            response.body("cases.size()", is(1));
+            response.body("cases[0].case_data." + field, is(expectedValue));
+        }
+    }
+
+    @AfterAll
+    void cleanUp() {
+        deleteIndexAndAlias();
+    }
+
+    private void createCases() {
+        createCaseAndProgressState(asPrivateCaseworker(true));
+        createCase(asPrivateCaseworker(true), FullCase.build());
+
+        // wait until logstash reads the case data
+        sleep(aat.getLogstashReadDelay());
```

Comment:
a sleep works fine when the number of test subclasses is small. As we add more, each of them will add a 5sec delay to the func tests...this can be an issue. 
An alternative strategy to remove this sleep, would be to retry a single test a certain number of times before considering it failed. For example it could be tried 3 times with a 2 sec delay between the first retry, and 1 sec delay between the 2nd and 3rd retry. The advantage with this approach is that if LS is fast to index, then we won't have to pay the fixed 5 sec penalty. 
It's more effort to implement this strategy though, even thought I'm sure there are libraries our there that handle a retry automatically.
I leave the decision to you @gbenadikar

Reply:
Can we not create cases as a pre step to Test? This way by the time build gets to actually running tests, the required time for logstash wouldve gone past and we will see the results without having to add sleep?

Reply:
@mario-paniccia , @ngul, I have introduced a test data loader extension that will import definition and create test cases only once for all the Elasticsearch test classes extending it. We can now have multiple test classes using extension to load data upfront before actual execution. Also lowered the default logstash delay to 2 seconds  as we are creating multiple test cases, Logstash should catch-up anyway.  It is also good to have control over the test data within the test framework.

Reply:
Any functional test that depends on components with a delay in reflecting test data needs to use the extension. Other functional tests for e.g. get case data, can create the case and then get it sequentially.